code/modules/vending/vending.dm

```diff
@@ -277,18 +278,23 @@ ADMIN_INTERACT_PROCS(/obj/machinery/vending, proc/throw_item)
 /obj/machinery/vending/blob_act(var/power)
 	if (prob(power * 1.25))
 		SPAWN(0)
-			if (prob(power / 3) && can_fall == 2)
+			if (prob(power / 3) &&  fallen)
 				for (var/i = 0, i < rand(4,7), i++)
 					src.malfunction()
 				qdel(src)
-			if (prob(50) || can_fall == 2)
+			if (prob(50) || fallen)
 				src.malfunction()
 			else
 				src.fall()
 		return
 
 	return
 
+/obj/machinery/vending/bullet_act(var/obj/projectile/P)
+	if((src.can_fall) && prob(P.power))
+		src.fall()
```

Comment:
should probably do checks on `ks_ratio` and/or `damage_type` - knocking over vending machines with a phaser would feel odd, and with a taser even odder

Reply:
Ah, gocha. Not too familar with how guns work - no problem restricting it to kinetics, nukies don't have energy weapons last I checked.

Reply:
Made a change so that guns that deal kinetic damage count (such as macrophasers) since those make sense

Reply:
If filtering by damage_type, should probably include `D_PIERCING` and `D_SLASHING`, (though the latter is... uncommon) to cover the full gamut

Reply:
Huh, didn't know they did those types of damage. Looking at the defines for projectile damage types I see:
#define D_KINETIC				(1<<0)
#define D_PIERCING				(1<<1)
#define D_SLASHING				(1<<2)
#define D_ENERGY				(1<<3)
#define D_BURNING				(1<<4)
#define D_RADIOACTIVE			(1<<5)
#define D_TOXIC					(1<<6)
#define D_SPECIAL				(1<<7)
D_KINETIC, D_PIERCING and D_SLASHING are the only ones of those that make sense yeah

I can't see a case where a weapon with that damage type stuns, even det revolver is D_ENERGY, so I'll add checks for only those two